File: src/types/scene.ts
export interface LevelSummary {
  levelId: string;
  name: string;
  path: string;
  relativePath: string;
  modifiedAt: string;
}

export interface SceneLevel {
  levelId: string;
  name: string;
  path: string;
  createdAt: string;
  updatedAt: string;
  layers: WorldLayer[];
  objects: SceneObject[];
}

export interface WorldLayer {
  id: string;
  name: string;
  visible: boolean;
  color: string;
}

export interface SceneObject {
  id: string;
  name: string;
  tags: string[];
  layer: string | null;
  visible: boolean;
  assetReference: string | null;
  transform: Transform | null;
  components: SceneComponent[];
}

export interface Transform {
  position: Vec3;
  rotation: Vec3;
  scale: Vec3;
}

export interface Vec3 {
  x: number;
  y: number;
  z: number;
}

export interface SceneComponent {
  componentType: string;
  data: Record<string, unknown>;
}

File: src-tauri/src/models/scene.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LevelSummary {
    pub level_id: String,
    pub name: String,
    pub path: String,
    pub relative_path: String,
    pub modified_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SceneLevel {
    pub level_id: String,
    pub name: String,
    pub path: String,
    pub created_at: String,
    pub updated_at: String,
    pub layers: Vec<WorldLayer>,
    pub objects: Vec<SceneObject>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorldLayer {
    pub id: String,
    pub name: String,
    pub visible: bool,
    pub color: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SceneObject {
    pub id: String,
    pub name: String,
    pub tags: Vec<String>,
    pub layer: Option<String>,
    pub visible: bool,
    pub asset_reference: Option<String>,
    pub transform: Option<Transform>,
    pub components: Vec<SceneComponent>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Transform {
    pub position: Vec3,
    pub rotation: Vec3,
    pub scale: Vec3,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Vec3 {
    pub x: f32,
    pub y: f32,
    pub z: f32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SceneComponent {
    pub component_type: String,
    pub data: serde_json::Value,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateLevelRequest {
    pub project_root: String,
    pub name: String,
}

